tests/test_timeseries_dataframe.py

```diff
@@ -0,0 +1,126 @@
+        'DATA_DELTA_VEL[3] ', 'DATA_DELTA_ANGLE[1] ',
+        'DATA_DELTA_ANGLE[2] ', 'DATA_DELTA_ANGLE[3]'
+    ]
+
+    columns_sol2 = [
+        'DATA_DELTA_VEL[1] ', 'DATA_DELTA_VEL[2] ',
+        'TIME_NANOSECONDS_TAI ', 'DATA_DELTA_VEL[3] ',
+        'DATA_DELTA_ANGLE[1] ', 'DATA_DELTA_ANGLE[2] ',
+        'DATA_DELTA_ANGLE[3]'
+    ]
+
+    def test_read_from_text_file_basic(self):
+        tsdf = TimeSeriesDataFrame()
+
+        tsdf.read_from_text_file(
+            os.path.join(self.DATADIR, "dlc_trimmed.csv")
+        )
+        self.assertEqual(tsdf._columns, self.columns_sol)
+        self.assertEqual(len(tsdf._columns), 6)
+        for i in range(len(tsdf._columns)):
+            self.assertEqual(tsdf._data[i].ndarray.shape[0], 10)
+        self.assertEqual(tsdf._index_column_name, 'TIME_NANOSECONDS_TAI ')
+
+        tsdf.read_from_text_file(
+            os.path.join(self.DATADIR, "dlc_trimmed_header_changed.csv"),
+            delimiter=',',
+            timestamp_column='TIME_NANOSECONDS_TAI '
+        )
+
+        self.assertEqual(tsdf._columns, self.columns_sol)
+        self.assertEqual(len(tsdf._columns), 6)
+        for i in range(len(tsdf._columns)):
+            self.assertEqual(tsdf._data[i].ndarray.shape[0], 10)
+        self.assertEqual(tsdf._index_column_name, 'TIME_NANOSECONDS_TAI ')
+
+        tsdf.read_from_text_file(
+            os.path.join(self.DATADIR, "dlc_trimmed_header_changed.csv"),
+            delimiter=',',
+            timestamp_in_file=False
+        )
+        self.assertEqual(tsdf._columns, self.columns_sol2)
+        self.assertEqual(len(tsdf._columns), 7)
+        for i in range(len(tsdf._columns)):
+            self.assertEqual(tsdf._data[i].ndarray.shape[0], 10)
+        self.assertEqual(tsdf._index_column_name, 'Index')
+
+    def test_dataframe_attribute_columns(self):
+        tsdf = TimeSeriesDataFrame()
+        tsdf.read_from_text_file(
+            os.path.join(self.DATADIR, "dlc_trimmed.csv")
+        )
+        self.assertEqual(tsdf.columns, self.columns_sol)
+
+    def test_dataframe_attribute_shape(self):
+        tsdf = TimeSeriesDataFrame()
+        tsdf.read_from_text_file(
+            os.path.join(self.DATADIR, "dlc_trimmed.csv")
+        )
+        self.assertEqual(tsdf.shape, (10, 6))
+
+    def test_dataframe_attribute_index(self):
+        tsdf = TimeSeriesDataFrame()
+        tsdf.read_from_text_file(
+            os.path.join(self.DATADIR, "dlc_trimmed.csv")
+        )
+
+        nd_arr = np.genfromtxt(
+            os.path.join(self.DATADIR, "dlc_trimmed.csv"), delimiter=','
+        )[1:]
+
+        self.assertEqual(
+            list(tsdf.index), list(nd_arr[:, 0].astype(np.uint64))
+        )
+
+    def test_dataframe_get_column(self):
+        tsdf = TimeSeriesDataFrame()
+        tsdf.read_from_text_file(
+            os.path.join(self.DATADIR, "dlc_trimmed.csv")
+        )
+
+        one_column_data = tsdf['DATA_DELTA_VEL[1] ']
```

Comment:
I’m wondering if the user needs to include a space when entering the column name?

Reply:
Hi @j8xixo12
My design is to preserve whatever there is in the CSV. I do not want to assume what the data "should be". If there is a space in the column name in the original file, I will read as is. This is also the behavior of Pandas if you use it to read my test CSVs.

Reply:
It is not user friendly to require them remember the whitespaces in text file.  It is especially difficult when the whitespaces are at the end of a line.  Trailing whitespaces are invisible in many editors and web browsers.

In this early prototype it's OK to keep the space in key names.  But I don't think it's a good design to force keeping whitespaces.  We need to think about it in the future.

Reply:
Hi @yungyuc 

I think we can discuss further when we have F2F time, and I still want to keep the trailing white spaces since user can access the list of columns through "columns" property. I think there could be scenario where user want to keep the column name as is. Let's say user compose an automatic script to read the column names from another file (maybe the config of the data source instruments) and the trailing spaces is there in the beginning. Under this scenario the user may want the column name stays the same.

Reply:
You can keep it for now.

Reply:
Do not mark conversation resolved.  Doing it makes it harder to track.